cmd/datamon/cmd/config.go

```diff
@@ -39,31 +65,24 @@ func (c *CLIConfig) setDatamonParams(flags *flagsT) {
 	}
 }
 
-func extraMsg(msg, about string, err error) string {
-	// provide extra explanation and guidance about the error
-	switch err {
-	case storagestatus.ErrInvalidResource:
-		return fmt.Sprintf("%s: please check that the config bucket %q is a valid gcs bucket", msg, about)
-	case storagestatus.ErrNotExists:
-		return fmt.Sprintf("%s: please check that the context has been created in your config", msg)
-	}
-	return msg
+// MarshalConfig produces a CLI config as a YAML document
+func (c *CLIConfig) MarshalConfig() ([]byte, error) {
+	return yaml.Marshal(c)
```

Comment:
same as when this was inlined:

```
switch configExt {
case ".yaml":
  o, err := <serialize-yaml>
default:
  wrapFatalln("implementation error", errors.New("unknown config format"))
```

to encode

> the extension constant agrees with serialization routine

Reply:
ok done with a if and with a dynamic extension check, since the target file may be overridden with DATAMON_CONFIG var (see example)

Reply:
ok, i see that the `configExt` constant has been replaced by a `configFileLocation` function